ipa_room_segmentation/common/src/adaboost_classifier.cpp

```diff
@@ -233,19 +233,19 @@ void AdaboostClassifier::segmentMap(const cv::Mat& map_to_be_labeled, cv::Mat& s
 		if (boost::filesystem::exists(boost::filesystem::path(filename_room)) == false)
 			boost::filesystem::copy_file(filename_room_default, filename_room);
 #if CV_MAJOR_VERSION == 2
-		room_boost_.load(filename_room.c_str());
+    room_boost_.load(filename_room.c_str());
 #else
-		room_boost_->load(filename_room.c_str());
+    room_boost_->load<cv::ml::Boost>(filename_room.c_str());
```

Comment:
We might need to make a distinction here between OpenCV 3.3 (?) for kinetic (without ```<cv::ml::Boost>```) and OpenCV 3.2 (?) for melodic (with ```<cv::ml::Boost>```, because ```boost->load()``` was not yet implemented and hence ```algorihtm->load<cv::ml::Boost>()``` was used).

This can be resolved by using
```
#if CV_MAJOR_VERSION == 2
		room_boost_.load(filename_room.c_str());
#elif CV_MAJOR_VERSION ==3 &&  CV_MINOR_VERSION<=2
		room_boost_.load<cv::ml::Boost>(filename_room.c_str());
#else
		room_boost_->load(filename_room.c_str());
#endif

I hope.

And please keep the tab indentation as is and do not mix in spaces into a file accurately indented with tabs (although you do not like tabs, I know, but please keep existing style as is).

Reply:
Since this is getting more complicated I extracted it to a separate function.
I have reverted to tab indentation for the files I edited.